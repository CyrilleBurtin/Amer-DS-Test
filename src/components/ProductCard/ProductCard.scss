.slm-product-card {
  @apply bg-white
    max-w-[354px];

  &__image {
    @apply w-[354px]
    h-[210px]
    object-cover
  }

  &__content {
    @apply p-6
      flex
      flex-col
      flex-grow
      justify-between;
  }

  &__header {
    @apply h-[138px];
  }

  &__info {
    @apply flex
      justify-between;
  }

  &__time {
    @apply flex
      items-center
      text-base
      gap-2;
  }
}
