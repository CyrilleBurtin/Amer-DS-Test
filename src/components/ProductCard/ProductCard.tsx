import { type Products } from '../../interfaces/products'
import './ProductCard.scss'

const ProductCard = ({ img, title, subtitle, time, label }: Products) => {
  const classes = {
    container: 'slm-product-card',
    image: 'slm-product-card__image'
  }

  return (
    <div className={classes.container}>
      <img src={img} alt={subtitle} className={classes.image} />
      fdsfsd
    </div>
  )
}

export default ProductCard
