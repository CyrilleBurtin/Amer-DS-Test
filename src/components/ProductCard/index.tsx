import Title from '../Title';
import Text from '../Text';
import Label from '../Label';
import { type Product } from '../../interfaces/product';
import { LABEL_VARIANT } from '../Label/constants';
import { TITLE_VARIANT } from '../Title/constants';
import timer from 'assets/icons/timer.svg';
import './ProductCard.scss';

const ProductCard = ({ img, title, description, time, label }: Product) => {
  const classes = {
    container: 'slm-product-card',
    image: 'slm-product-card__image',
    content: 'slm-product-card__content',
    header: 'slm-product-card__header',
    info: 'slm-product-card__info',
    time: 'slm-product-card__time'
  };

  return (
    <div className={classes.container}>
      <img src={`${img}`} alt={description} width="354" height="210" />
      <div className={classes.content}>
        <header className={classes.header}>
          <Title is="h2" variant={TITLE_VARIANT.RED}>
            {title}
          </Title>
          <Text>{description}</Text>
        </header>
        <div className={classes.info}>
          <p className={classes.time}>
            <img
              src={timer}
              alt="timing"
              width="24"
              height="24"
              loading="lazy"
            />{' '}
            {time}
            &nbsp;min
          </p>
          <Label variant={LABEL_VARIANT.GRAY}>{label}</Label>
        </div>
      </div>
    </div>
  );
};

export default ProductCard;
