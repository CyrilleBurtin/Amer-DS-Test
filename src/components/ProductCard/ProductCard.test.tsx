import { render } from '@testing-library/react'
import ProductCard from './'
import { describe } from 'vitest'

describe('<ProductCard>', () => {
  const product = {
    id: '12345ABD',
    img: 'image/test.jpeg',
    title: 'Ski de rando',
    description: 'Pour les skieur confirmés',
    time: 10,
    label: 'Promo'
  }

  it('should render the component', () => {
    const { container, getByText } = render(<ProductCard {...product} />)

    expect(getByText(/Ski de rando/i)).toBeInTheDocument()
    expect(getByText(/Pour les skieur confirmés/i)).toBeInTheDocument()
    expect(getByText(/Promo/i)).toBeInTheDocument()

    expect(container).toMatchSnapshot()
  })
})
