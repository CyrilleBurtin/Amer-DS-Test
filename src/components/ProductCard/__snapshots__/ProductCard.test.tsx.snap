// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<ProductCard> > should render the component and create a snapshot 1`] = `
<div>
  <div
    class="slm-product-card"
  >
    <img
      alt="Pour les skieur confirmés"
      class="slm-product-card__image"
      src="image/test.jpeg"
    />
    <div
      class="slm-product-card__content"
    >
      <header
        class="slm-product-card__header"
      >
        <h2
          class="slm-title --red"
        >
          Ski de rando
        </h2>
        <p
          class="slm-text"
        >
          Pour les skieur confirmés
        </p>
      </header>
      <div
        class="slm-product-card__info"
      >
        <p
          class="slm-product-card__time"
        >
          <img
            alt="timing"
            height="24"
            loading="lazy"
            src="/src/assets/icons/timer.svg"
            width="24"
          />
           
          10
           min
        </p>
        <span
          class="slm-label --gray"
        >
          Promo
        </span>
      </div>
    </div>
  </div>
</div>
`;
